pkg/cmd/server/api/types.go

```diff
@@ -539,6 +542,25 @@ type MasterClients struct {
 	OpenShiftLoopbackKubeConfig string
 	// ExternalKubernetesKubeConfig is a .kubeconfig filename for proxying to kubernetes
 	ExternalKubernetesKubeConfig string
+
+	// OpenShiftLoopbackClientConnectionOverrides specifies client overrides for system components to loop back to this master.
+	OpenShiftLoopbackClientConnectionOverrides *ClientConnectionOverrides
+	// ExternalKubernetesClientConnectionOverrides specifies client overrides for proxying to Kubernetes.
+	ExternalKubernetesClientConnectionOverrides *ClientConnectionOverrides
+}
+
+type ClientConnectionOverrides struct {
```

Comment:
I don't really like this being all or nothing and we have to live with this config forever.

@liggitt do you object too or am I just being ornery?